### ISSUE TITLE
Error: [Express Route Fake Error] Action for method 'post' with route '/' does not exist

### ISSUE THREAD

**Issue by cjduncana** [cmstead/express-route-fake#2]
# TL;DR

When the server receives a POST request to the index page, it sends a 404 HTTP response. I expect that when I write the test, express-route-fake would behave similar to Express.js' Routers and respond with a 404 HTTP response, instead it throws an error. How can I test for a 404 HTTP response, or at least test that there is no router for POST requests?
## Problem

I have this router that responds with a rendered page and a 200 HTTP response when requested by a GET request.
### controllers.js

``` javascript
var router = require('express').Router();

router.get('/', function(request, response) {
    response.status(200).render('index');
});

module.exports = router;
```
### test.js The Test

``` javascript
it('Post to the Index Page', function(done) {
    var routeAction = expressRouteFake.getRouteAction('post', '/');
    routeAction(req, res);
    res.resData.status.should.be.equal(404);
    return done();
});
```

The rest of the code for the test is almost identical to the one written in [your article](http://www.chrisstead.com/archives/1128/unit-testing-express-routing/). As for this test, when the response returns, check the status code. It should be a 404 HTTP status code.
### Run `npm test`

I run the test, but I get the following error:

```
1) Testing the Index Page Controllers: Post to the Index Page:
     Error: [Express Route Fake Error] Action for method 'post' with route '/' does not exist
```

I do not know why, and I do not understand why either. Could you help me with this?

**Comment by cmstead** [comment]
This error is due to the fact that there is no post route defined. Express
will handle missing routes without intervention.

If you want to guarantee that a post to / is disallowed, you can write a
test which checks that no function exists at for post at the / endpoint.

I hope that helps

**Comment by cjduncana** [comment]
I have tried `should.not.exist(expressRouteFake.routeFunctions.post);` and `expressRouteFake.routeFunctions.post.should.be.empty();`, but they both stated an error `TypeError: Cannot read property 'post' of undefined`. It seems that I do not have access to the route functions, that a pointer to them or a function that retrieves them is not being exported.

I modified `express-route-fake` to export the `routeFunctions` object and then tested with `expressRouteFake.routeFunctions.post.should.be.empty();` and it works, but then I realize that it is not pointing to the same object the other functions point to.

**Comment by cmstead** [comment]
Try this:

should.not.exist(expressRouteFake.getRouteAction('post', '/'));

RouteFunctions is not exposed to limit direct manipulation.

**Comment by cjduncana** [comment]
I still get:

```
1) Testing the Index Page Controllers: Post to the Index Page:
     Error: [Express Route Fake Error] Action for method 'post' with route '/' does not exist
      at throwOnUndefined (node_modules/express-route-fake/index.js:16:15)
      at throwOnBadKeys (node_modules/express-route-fake/index.js:27:5)
      at Object.getRouteAction (node_modules/express-route-fake/index.js:31:5)
      at Context.<anonymous> (controllers.js:37:43)
```

**Comment by cmstead** [comment]
I just reviewed the module source and I forgot I have it throw on undefined routes so you can identify when a route was not defined.  The error actually spells out the reason it is thrown specifically so you can read it and understand why a route function was not found.  Just verify that it throws:

should.throw(expressRouteFake.getRouteAction('post', '/'));

I'll spell that out in the readme for future users.

**Comment by cmstead** [comment]
Readme is updated with latest API information. I think this should help for other users in the future.

**Comment by cjduncana** [comment]
I tried both `should.throw(expressRouteFake.getRouteAction('post', '/'));` and `expressRouteFake.getRouteAction('post', '/').should.throw();`. They return the same error.

**Comment by cjduncana** [comment]
I wrote the following mock function:

``` javascript
(function() {
    throw new Error('Boo');
}).should.throw(Error);
```

It works, so the problem is not in Should. Without the Error class as a parameter, `should.throw()` let the error go through.

I tried the same with your code:

``` javascript
(expressRouteFake.getRouteAction('post', '/')).should.throw(Error);
```

The error still manifests.
